bulldozer/merge.go

```diff
@@ -36,13 +36,24 @@ func MergePR(ctx context.Context, pullCtx pull.Context, client *github.Client, m
 
 	mergeOpts := &github.PullRequestOptions{}
 
-	switch mergeConfig.Method {
-	case SquashAndMerge, MergeCommit, RebaseAndMerge:
-		mergeOpts.MergeMethod = string(mergeConfig.Method)
-	default:
-		mergeOpts.MergeMethod = string(MergeCommit)
+	base, _, err := pullCtx.Branches(ctx)
+	if err != nil {
+		logger.Error().Msg("Unable to find the base branch. Aborting.")
+		return err
+	}
+
+	var mergeMethod = mergeConfig.Method
+
+	if baseMergeOption, ok := mergeConfig.BranchMergeMethod[base]; ok {
```

Comment:
style: I'd call this `branchMergeMethod` or `branchMethod` instead of `baseMergeOption` for consistency with other variables